cmd/gardener-admission-controller/app/app.go

```diff
@@ -124,3 +131,16 @@ func run(ctx context.Context, log logr.Logger, cfg *admissioncontrollerconfigv1a
 	log.Info("Starting manager")
 	return mgr.Start(ctx)
 }
+
+func addAllFieldIndexes(ctx context.Context, i client.FieldIndexer) error {
+	for _, fn := range []func(context.Context, client.FieldIndexer) error{
+		// core API group
+		indexer.AddShootStatusUID,
+	} {
+		if err := fn(ctx, i); err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
Not sure if I am missing something, but is this not a very convoluted way of doing the same as below? If so why not keep it simple?

```suggestion
	return indexer.AddShootStatusUID(ctx, i)
```

Reply:
We might add other index functions as well, and I already want to define the style so that it'll be similar to our other binaries: https://github.com/gardener/gardener/blob/868185e021c33c822cfe43f157fdacd8897ccc8d/cmd/gardener-controller-manager/app/app.go#L145-L167